samples/istio/service-to-service/README.md

```diff
@@ -0,0 +1,198 @@
+  
+
+- Add an `/etc/host` entry for Choreo Connect and WSO2 API Manager
+
+    ```
+    127.0.0.1  apim.wso2.com gw.wso2.com
+    ```
+
+### 3. Deploy and Apply API Management for the Trains Microservice
+
+- Deploy the **Trains** service.
+  ```sh
+  kubectl apply -f microservices/trains.yaml
+  ```
+
+- Access the Publisher portal - https://apim.wso2.com/publisher/. Use `admin:admin` as the credentials.
+
+- Use [trains-openapi.yaml](trains-openapi.yaml) to create the API. Publisher portal will extract following values from the open API yaml. Let's keep those values and create the API.
+  ```
+  API Name: Trains
+  Context: trains
+  Version: 1.0.0
+  Endpoint: http://trains/trains-service/v1
+  ```
+
+- [Enable API key security](https://apim.docs.wso2.com/en/4.1.0/design/api-security/api-authentication/secure-apis-using-api-keys/#using-api-keys-to-secure-an-api) for the created API.
+
+- Deploy and Publish the API.
+
+- Access the Dev portal - https://apim.wso2.com/devportal/. Use `admin:admin` as the credentials. Subscribe to an application and get an API Key with **infinite validity period**. Let's assign it to the bash variable `API_KEY` as follows.
+  ```sh
+  API_KEY=<API_KEY>
+  ```
+
+- Try out in DevPortal with API Key or by executing the following curl command.
+  ```sh
+  curl -X 'GET' \
+    "https://gw.wso2.com/trains/1.0.0/trains" \
+    -H "accept: application/json" \
+    -H "apikey: $API_KEY" -k
+  ```
+
+### 5. Deploy the Schedules Microservice
+
+- Create a Kubernetes secret with API Key of **Trains** API.
+  ```sh
+  kubectl create secret generic schedules-creds --from-literal=trains_service_api_key=$API_KEY
+  ```
+
+- Create a Virtual Service named `choreo-connect-internal` to set header `host: gw.wso2.com` (to set the Vhost of the gateway environment).
+  ```sh
+  kubectl apply -f cc-internal-vs.yaml
+  ```
+
+  > **NOTE:**
+  >
+  > We have used the Virtual Service `choreo-connect-internal` to set the header `host: gw.wso2.com`, hence the Virtual Service routes requests to the HTTP port of the Choreo Connect gateway. You can also use the HTTPS port of Choreo Connect and set the host header from the microservice (backend).
+
+- Deploy **Schedules** service.
+  ```sh
+  kubectl apply -f microservices/schedules.yaml
+  ```
+
+  Within the above Kubernetes Deployment we have set the following environment variables. Schedules microservice uses `TRAINS_SERVICE_URL` as the service URL of Trains microservice and hence we assigned `choreo-connect-internal`. Then requests will route through the Choreo Connect gateway. Schedules microservice includes the value of `TRAINS_SERVICE_API_KEY` in the header `apikey`.
+
+  ```yaml
+  env:
+    - name: TRAINS_SERVICE_URL
+      value: http://choreo-connect-internal/trains/1.0.0
+    - name: TRAINS_SERVICE_API_KEY
+      valueFrom:
+        secretKeyRef:
+          name: "schedules-creds"
+          key: "trains_service_api_key"
```

Comment:
Do we have steps to update the K8s secret with the API Key?

Reply:
moved the secret create command to here with https://github.com/wso2/product-microgateway/pull/2971/commits/28a5ff3f67b2b8073a815200b1eada7a7dc40b8f